services/history/src/models/repository.ts

```diff
@@ -0,0 +1,22 @@
+import mongoose from 'mongoose';
+import config from '../config';
+import { HistoryModel, HistoryStatus, IdType, Question, User } from './historyModel';
+
+export async function connectToDB() {
+    await mongoose.connect(config.DB_URI);
+}
+
+export async function createHistory(roomId: IdType, user1: User, user2: User, question: Question) {
+    return await HistoryModel.insertMany([
+        { roomId, user: user1, collaborator: user2, question },
+        { roomId, user: user2, collaborator: user1, question },
+    ]);
+}
+
+export async function updateHistory(roomId: IdType, userId: IdType, status: HistoryStatus) {
+    return await HistoryModel.findOneAndUpdate({ roomId, 'user._id': userId }, { $set: { status } }, { new: true });
```

Comment:
User1 can forfeit while User2 can continue on with coding, so we had to separate it out with isForfeit status (according to the user) and roomStatus to indicate whether the room still has a remaining user. I guess the data representation is not so good now that I think about it😅😅as you'll have to check if room is close and check for forfeits at the same time to tell whether the room was closed due to forfeit or successful submission. 
Eg; user1-> isForfeit = true, user2 -> isForfeit = true and room is closed would mean that the match ended because both users forfeited.

Reply:
yep, one user could forfeit and the other could submit. A lil funny but I think it does make sense